http/handler.go

```
@@ -111,32 +110,31 @@ func (s *Service) handleKeyRequest(w http.ResponseWriter, r *http.Request) {
 
 // handleTransaction
 func (s *Service) handleTransaction(w http.ResponseWriter, r *http.Request) {
-	// ...so we convert it to a string by passing it through
-	// a buffer first. A 'costly' but useful process.
+	var msg string
 	cmds := &raftpb.RaftCommand{}
 	if m, err := ioutil.ReadAll(r.Body); err != nil {
 		w.WriteHeader(http.StatusBadRequest)
-		return
+		msg = fmt.Sprintf("failed to read %v", r.Body)
 	} else if err = proto.Unmarshal(m, cmds); err != nil {
 		w.WriteHeader(http.StatusBadRequest)
-	}
-	resultCmds, err := s.coordinator.Transaction(cmds)
-	if err != nil {
-		s.log.Error(err)
+		msg = fmt.Sprintf("failed to parse %v", r.Body)
+	} else if resultCmds, err := s.coordinator.Transaction(cmds); err != nil {
```

Too many if/else s here. Let's fail early and return the error.

---

It's actually the same but left you with more lines of code.
It just changes 
```
if cond {
 ....
} else if {
 ....
} else {
 ....
}
return
```
to 
```
if {
...
 return
}
if {
...
 return
}
if {
...
 return
}
```